static/js/stream_create.js

```diff
@@ -307,99 +330,71 @@ export function show_new_stream_modal() {
         $("#announce-new-stream").hide();
     }
     clear_error_display();
-
-    $("#stream-checkboxes label.checkbox").on("change", function (e) {
-        const elem = $(this);
-        const stream_id = Number.parseInt(elem.attr("data-stream-id"), 10);
-        const checked = elem.find("input").prop("checked");
-        const subscriber_ids = new Set(peer_data.get_subscribers(stream_id));
-
-        $("#user-checkboxes label.checkbox").each(function () {
-            const user_elem = $(this);
-            const user_id = Number.parseInt(user_elem.attr("data-user-id"), 10);
-
-            if (subscriber_ids.has(user_id)) {
-                user_elem.find("input").prop("checked", checked);
-            }
-        });
-
-        e.preventDefault();
-    });
 }
 
 export function create_handlers_for_users(container) {
     // container should be $('#people_to_add')...see caller to verify
-    container.on("change", "#user-checkboxes input", update_announce_stream_state);
-
-    // 'Check all' and 'Uncheck all' visible users
-    container.on("click", ".subs_set_all_users", (e) => {
-        $("#user-checkboxes .checkbox").each((idx, li) => {
-            if (li.style.display !== "none") {
-                $(li.firstElementChild).prop("checked", true);
+    function update_checked_state_for_users(value, users) {
+        // Only do full_redraw if we have to render more than 500 users.
+        const full_redraw = !users || users.length > 500;
+        for (const user of all_users) {
+            // We don't want to uncheck the user creating the stream
+            // if it is not admin.
+            if (user.user_id === page_params.user_id && value === false && !page_params.is_admin) {
+                continue;
             }
-        });
-        e.preventDefault();
-        update_announce_stream_state();
-    });
+            // We update for all users if `users` parameter
+            // is empty.
+            if (!users || users.includes(user.user_id)) {
```

Comment:
Is this the right behavior?  I'm not sure you'll have the right handling if a user toggles a stream that somehow has no subscribers.

Better might be to check `users === undefined` so an empty list doesn't count?